buffer/src/main/java/io/netty5/buffer/adapt/AdaptivePoolingAllocator.java

```diff
@@ -241,6 +290,22 @@ public void close() {
         } finally {
             magazineExpandLock.unlockWrite(magsExpandWriteLock);
         }
+        if (liveCachedMagazines != null) {
+            liveCachedMagazines.forEach(mag -> {
+                try {
+                    mag.ownerEventExecutor.execute(() -> {
+                        threadLocalMagazine.remove();
+                        // this is to prevent that a subsequent request will try to create a new Magazine
+                        threadLocalMagazine.set(NO_MAGAZINE);
```

Comment:
We don't need to set this because there's a check on `closed` in `allocate()` before we touch the thread-local.